# Import Libraries

In [142]:
import numpy as np
import pandas as pd
pd.set_option("max_columns", None)
from scipy.stats import chi2_contingency
from sklearn.linear_model import Lasso
from sklearn.linear_model import LassoCV
from sklearn.linear_model import Ridge
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score

# Import Data

In [143]:
data = pd.read_csv('fashion_data_on_Instagram.csv')
data.set_index('UserId', inplace = True)
data.head()

Followings  Followers  MediaCount  \
UserId                                                              
1171579752855683619_212070047        518.0     9840.0         541   
1171594777274371222_176762322       7333.0     2300.0         272   
1171407552643586413_581125501        131.0      605.0         106   
1171407552643586413_581125501        131.0      605.0         106   
1171508187966229230_2797323089       132.0      456.0          56   

                                  BrandName BrandCategory  \
UserId                                                      
1171579752855683619_212070047   abercrombie   High street   
1171594777274371222_176762322   abercrombie   High street   
1171407552643586413_581125501   abercrombie   High street   
1171407552643586413_581125501   abercrombie   High street   
1171508187966229230_2797323089  abercrombie   High street   

                                                                         Hashtags  \
UserId                                                                              
1171579752855683619_212070047   beautiful, summer, fashion, love, cute, food, ...   
1171594777274371222_176762322   teen, model, brunette, selfie, hollister, snap...   
1171407552643586413_581125501   alexandani, cute, llbean, beanboots, anthropol...   
1171407552643586413_581125501   alexandani, cute, llbean, beanboots, anthropol...   
1171508187966229230_2797323089  abercrombieandfitch, shopping, love, hollister...   

                                                                          Caption  \
UserId                                                                              
1171579752855683619_212070047     We were born to be REAL, not to be perfect. ...   
1171594777274371222_176762322     Gotta run, but first, let me take selfie. Me...   
1171407552643586413_581125501                         dress to impress\U0001f457    
1171407552643586413_581125501                         dress to impress\U0001f457    
1171508187966229230_2797323089    #love #shopping #shoppen #hollister #abercro...   

                                                                           ImgURL  \
UserId                                                                              
1171579752855683619_212070047   https://scontent.cdninstagram.com/t51.2885-15/...   
1171594777274371222_176762322   https://scontent.cdninstagram.com/t51.2885-15/...   
1171407552643586413_581125501   https://scontent.cdninstagram.com/t51.2885-15/...   
1171407552643586413_581125501   https://scontent.cdninstagram.com/t51.2885-15/...   
1171508187966229230_2797323089  https://scontent.cdninstagram.com/t51.2885-15/...   

                                Likes  Comments   CreationTime  \
UserId                                                           
1171579752855683619_212070047      97          0    1453883211   
1171594777274371222_176762322      94          0    1453885002   
1171407552643586413_581125501      91          3    1453862683   
1171407552643586413_581125501      94          3    1453862683   
1171508187966229230_2797323089      9          1    1453874680   

                                                                    Link  \
UserId                                                                     
1171579752855683619_212070047   https://www.instagram.com/p/BBCSZPzAMIj/   
1171594777274371222_176762322   https://www.instagram.com/p/BBCVz4YJEyW/   
1171407552643586413_581125501   https://www.instagram.com/p/BBBrPZ2h9lt/   
1171407552643586413_581125501   https://www.instagram.com/p/BBBrPZ2h9lt/   
1171508187966229230_2797323089  https://www.instagram.com/p/BBCCH1zMEru/   

                                  Selfie  BodySnap  Marketing  ProductOnly  \
UserId                                                                       
1171579752855683619_212070047   0.000885  0.336438   0.741220     0.009454   
1171594777274371222_176762322   0.003912  0.995503   0.001728     0.002125   
1171407552643586413_5811

# Clean Data

### Let's See if there are any NaNs in the data

In [17]:
set(data['Hashtags'].isna())

{False, True}

In [18]:
set(data['Caption'].isna())

{False, True}

In [50]:
set(data['Comments '].isna())

{False}

##### So there exist NaN values in Hashtags and Captions. There are no other columns that contain NaN.

In [55]:
data.isna().groupby('Hashtags')['Followings'].count()

Hashtags
False    24606
True       146
Name: Followings, dtype: int64

##### There are 146 NaN rows in hashtags. It is possible that this is simply due to the fact that some posts do not have hashtags.

In [57]:
data.isna().groupby('Caption')['Followings'].count()

Caption
False    24717
True        35
Name: Followings, dtype: int64

##### There are 35 NaN rows in Caption

### Maybe the caption or hashtag information is missing because there is either only a caption or only hashtags....

In [69]:
hashtag_nan = list(data['Hashtags'].isna())
caption_nan = list(data['Caption'].isna())

In [73]:
count = 0
for i in range(len(hashtag_nan)):
    if hashtag_nan[i] and caption_nan[i]:
        count += 1
print('The number of times both hashtag_nan and caption_nan is true is', count)

The number of times both hashtag_nan and caption_nan is true is 18


##### So we have missing hashtag both 18/35 times that we have a missing caption. So it is not exactly correlated. Maybe we can just drop the rows that have missing caption or hashtag if we need to use these, or claim that there was not hashtag or caption for these rows.

### I want to try and see if I can make brand category numeric

In [144]:
brand_category = pd.get_dummies(data['BrandCategory'])

In [145]:
brand_category.head()

Designer  High street  Mega couture  \
UserId                                                                
1171579752855683619_212070047          0            1             0   
1171594777274371222_176762322          0            1             0   
1171407552643586413_581125501          0            1             0   
1171407552643586413_581125501          0            1             0   
1171508187966229230_2797323089         0            1             0   

                                Small couture  
UserId                                         
1171579752855683619_212070047               0  
1171594777274371222_176762322               0  
1171407552643586413_581125501               0  
1171407552643586413_581125501               0  
1171508187966229230_2797323089              0

In [146]:
data_new = pd.concat([data, brand_category], sort = False, axis = 1)
data_new.head()

Followings  Followers  MediaCount  \
UserId                                                              
1171579752855683619_212070047        518.0     9840.0         541   
1171594777274371222_176762322       7333.0     2300.0         272   
1171407552643586413_581125501        131.0      605.0         106   
1171407552643586413_581125501        131.0      605.0         106   
1171508187966229230_2797323089       132.0      456.0          56   

                                  BrandName BrandCategory  \
UserId                                                      
1171579752855683619_212070047   abercrombie   High street   
1171594777274371222_176762322   abercrombie   High street   
1171407552643586413_581125501   abercrombie   High street   
1171407552643586413_581125501   abercrombie   High street   
1171508187966229230_2797323089  abercrombie   High street   

                                                                         Hashtags  \
UserId                                                                              
1171579752855683619_212070047   beautiful, summer, fashion, love, cute, food, ...   
1171594777274371222_176762322   teen, model, brunette, selfie, hollister, snap...   
1171407552643586413_581125501   alexandani, cute, llbean, beanboots, anthropol...   
1171407552643586413_581125501   alexandani, cute, llbean, beanboots, anthropol...   
1171508187966229230_2797323089  abercrombieandfitch, shopping, love, hollister...   

                                                                          Caption  \
UserId                                                                              
1171579752855683619_212070047     We were born to be REAL, not to be perfect. ...   
1171594777274371222_176762322     Gotta run, but first, let me take selfie. Me...   
1171407552643586413_581125501                         dress to impress\U0001f457    
1171407552643586413_581125501                         dress to impress\U0001f457    
1171508187966229230_2797323089    #love #shopping #shoppen #hollister #abercro...   

                                                                           ImgURL  \
UserId                                                                              
1171579752855683619_212070047   https://scontent.cdninstagram.com/t51.2885-15/...   
1171594777274371222_176762322   https://scontent.cdninstagram.com/t51.2885-15/...   
1171407552643586413_581125501   https://scontent.cdninstagram.com/t51.2885-15/...   
1171407552643586413_581125501   https://scontent.cdninstagram.com/t51.2885-15/...   
1171508187966229230_2797323089  https://scontent.cdninstagram.com/t51.2885-15/...   

                                Likes  Comments   CreationTime  \
UserId                                                           
1171579752855683619_212070047      97          0    1453883211   
1171594777274371222_176762322      94          0    1453885002   
1171407552643586413_581125501      91          3    1453862683   
1171407552643586413_581125501      94          3    1453862683   
1171508187966229230_2797323089      9          1    1453874680   

                                                                    Link  \
UserId                                                                     
1171579752855683619_212070047   https://www.instagram.com/p/BBCSZPzAMIj/   
1171594777274371222_176762322   https://www.instagram.com/p/BBCVz4YJEyW/   
1171407552643586413_581125501   https://www.instagram.com/p/BBBrPZ2h9lt/   
1171407552643586413_581125501   https://www.instagram.com/p/BBBrPZ2h9lt/   
1171508187966229230_2797323089  https://www.instagram.com/p/BBCCH1zMEru/   

                                  Selfie  BodySnap  Marketing  ProductOnly  \
UserId                                                                       
1171579752855683619_212070047   0.000885  0.336438   0.741220     0.009454   
1171594777274371222_176762322   0.003912  0.995503   0.001728     0.002125   
1171407552643586413_5811

##### Now the brand category is represented in the last four columns of the dataframe with one_hot_encoding.

### In order to model with these we need to normalize the variables

In [9]:
for col in data_new.columns:
    if col not in ['BrandName', 'BrandCategory','Hashtags', 'Caption', 'ImgURL', 'Likes', 'Comments ', 'CreationTime','Link','Designer',
       'High street', 'Mega couture', 'Small couture']:
        data_new[col] = data_new[col]/np.linalg.norm(np.array(data_new[col], dtype = float))

# Model with Numeric Data Only

First, lets try to build a model with only the numeric data. This will be easy to do and will allow us to understand whether any of these numeric factors influence the number of likes/comments a user gets. 

### Likes Model

In [71]:
X = data_new[['Followings', 'Followers', 'MediaCount', 'Selfie', 'BodySnap', 'Marketing', 'ProductOnly', 'NonFashion',
       'Face', 'Logo', 'BrandLogo', 'Smile', 'Outdoor', 'NumberOfPeople',
       'NumberOfFashionProduct', 'Anger', 'Contempt', 'Disgust', 'Fear',
       'Happiness', 'Neutral', 'Sadness', 'Surprise', 'Designer',
       'High street', 'Mega couture', 'Small couture']]
y = data_new['Likes']

In [72]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

Let's use a Lasso Regression model to regress based on the likes that we receive. I want to use lasso because lasso will send the unimportant variables to zero allowing us to see which features are important. 

In [152]:
lasso = Lasso(alpha = 0.0001, max_iter = 10000).fit(X_train, y_train)

In [153]:
predictions = lasso.predict((X_test))
r2_score(y_test, predictions)

0.2984467039649339

Maybe ridge regression will work better?

In [180]:
ridge = Ridge(alpha = 0.0001).fit(X_train, y_train)

In [181]:
predictions = ridge.predict((X_test))
r2_score(y_test, predictions)

0.29844937687206075

So ridge regression is really no different...

### Try a Deep Learning Model

In [73]:
from keras.models import Sequential
from keras.layers import Dense
from keras.wrappers.scikit_learn import KerasRegressor
def our_model():
    model = Sequential()

    model.add(Dense(units=64, activation='relu', input_dim = 27))
    model.add(Dense(units=64, activation='relu', input_dim = 27))
    model.add(Dense(units=64, activation='relu', input_dim = 27))
    model.add(Dense(units=1))
    model.compile(loss='mean_squared_error', optimizer='adam')
    return model

In [74]:
estimator = KerasRegressor(build_fn=our_model, epochs=100, batch_size=5, verbose=1)

In [75]:
from sklearn.model_selection import KFold, cross_val_score

kfold = KFold(n_splits=10)
results = cross_val_score(estimator, X, y, cv=kfold)
print("Baseline: %.2f (%.2f) MSE" % (results.mean(), results.std()))

Epoch 1/100
22276/22276 [==============================] - 3s 144us/step - loss: 3987850.1403
Epoch 2/100
22276/22276 [==============================] - 3s 139us/step - loss: 896006.7696
Epoch 3/100
22276/22276 [==============================] - 3s 135us/step - loss: 46557687.4451
Epoch 4/100
22276/22276 [==============================] - 3s 134us/step - loss: 1540440.2986
Epoch 5/100
22276/22276 [==============================] - 3s 134us/step - loss: 1032275.9685
Epoch 6/100
22276/22276 [==============================] - 3s 134us/step - loss: 2056608.9093
Epoch 7/100
22276/22276 [==============================] - 3s 137us/step - loss: 1148519.9571
Epoch 8/100
16325/22276 [====================>.........] - ETA: 0s - loss: 1249694.7402

KeyboardInterrupt: 

Maybe for the neural network model to work we need to normalize the outputs as well??? I have only normalized the input space. Maybe we need to investigate how the input space was normalized. Maybe the categorical variables that are represented by 1's and 0's at the end of the dataframe need to be normalized?

Average loss value for Keras deep learning model based on just numeric values and predicting likes is 154K.

Maybe try using likes / followers?

# Likes to Follower Ratio

First check if there are any null likes or follower entries

In [198]:
data['Likes'].isnull().any() 

False

In [199]:
data['Followers'].isnull().any()

False

In [335]:
data_ver2 = pd.concat([data, brand_category], sort = False, axis = 1)
data_ver2['LikesToFollowers'] = data_ver2['Likes'] / data_ver2['Followers']
data_ver2.head()

Followings  Followers  MediaCount  \
UserId                                                              
1171579752855683619_212070047        518.0     9840.0         541   
1171594777274371222_176762322       7333.0     2300.0         272   
1171407552643586413_581125501        131.0      605.0         106   
1171407552643586413_581125501        131.0      605.0         106   
1171508187966229230_2797323089       132.0      456.0          56   

                                  BrandName BrandCategory  \
UserId                                                      
1171579752855683619_212070047   abercrombie   High street   
1171594777274371222_176762322   abercrombie   High street   
1171407552643586413_581125501   abercrombie   High street   
1171407552643586413_581125501   abercrombie   High street   
1171508187966229230_2797323089  abercrombie   High street   

                                                                         Hashtags  \
UserId                                                                              
1171579752855683619_212070047   beautiful, summer, fashion, love, cute, food, ...   
1171594777274371222_176762322   teen, model, brunette, selfie, hollister, snap...   
1171407552643586413_581125501   alexandani, cute, llbean, beanboots, anthropol...   
1171407552643586413_581125501   alexandani, cute, llbean, beanboots, anthropol...   
1171508187966229230_2797323089  abercrombieandfitch, shopping, love, hollister...   

                                                                          Caption  \
UserId                                                                              
1171579752855683619_212070047     We were born to be REAL, not to be perfect. ...   
1171594777274371222_176762322     Gotta run, but first, let me take selfie. Me...   
1171407552643586413_581125501                         dress to impress\U0001f457    
1171407552643586413_581125501                         dress to impress\U0001f457    
1171508187966229230_2797323089    #love #shopping #shoppen #hollister #abercro...   

                                                                           ImgURL  \
UserId                                                                              
1171579752855683619_212070047   https://scontent.cdninstagram.com/t51.2885-15/...   
1171594777274371222_176762322   https://scontent.cdninstagram.com/t51.2885-15/...   
1171407552643586413_581125501   https://scontent.cdninstagram.com/t51.2885-15/...   
1171407552643586413_581125501   https://scontent.cdninstagram.com/t51.2885-15/...   
1171508187966229230_2797323089  https://scontent.cdninstagram.com/t51.2885-15/...   

                                Likes  Comments   CreationTime  \
UserId                                                           
1171579752855683619_212070047      97          0    1453883211   
1171594777274371222_176762322      94          0    1453885002   
1171407552643586413_581125501      91          3    1453862683   
1171407552643586413_581125501      94          3    1453862683   
1171508187966229230_2797323089      9          1    1453874680   

                                                                    Link  \
UserId                                                                     
1171579752855683619_212070047   https://www.instagram.com/p/BBCSZPzAMIj/   
1171594777274371222_176762322   https://www.instagram.com/p/BBCVz4YJEyW/   
1171407552643586413_581125501   https://www.instagram.com/p/BBBrPZ2h9lt/   
1171407552643586413_581125501   https://www.instagram.com/p/BBBrPZ2h9lt/   
1171508187966229230_2797323089  https://www.instagram.com/p/BBCCH1zMEru/   

                                  Selfie  BodySnap  Marketing  ProductOnly  \
UserId                                                                       
1171579752855683619_212070047   0.000885  0.336438   0.741220     0.009454   
1171594777274371222_176762322   0.003912  0.995503   0.001728     0.002125   
1171407552643586413_5811

In [213]:
data_test = data_ver2[data_ver2['LikesToFollowers'].isnull()]
data_test.head()

Followings  Followers  MediaCount  \
UserId                                                              
1172628647727092971_2860928604         1.0        0.0           5   
1170996282170475597_2603666216         6.0        0.0           3   

                                       BrandName BrandCategory  \
UserId                                                           
1172628647727092971_2860928604         hollister   High street   
1170996282170475597_2603666216  marcbymarcjacobs      Designer   

                                                                         Hashtags  \
UserId                                                                              
1172628647727092971_2860928604            hollister, discountclothes, hollisterco   
1170996282170475597_2603666216  marcjacobs, marcbymarcjacobs, watch, love, loveit   

                                                                          Caption  \
UserId                                                                              
1172628647727092971_2860928604    Hollister pink and white striped sweater. Ne...   
1170996282170475597_2603666216    #marcjacobs #marcbymarcjacobs #love #loveit ...   

                                                                           ImgURL  \
UserId                                                                              
1172628647727092971_2860928604  https://scontent.cdninstagram.com/t51.2885-15/...   
1170996282170475597_2603666216  https://scontent.cdninstagram.com/t51.2885-15/...   

                                Likes  Comments   CreationTime  \
UserId                                                           
1172628647727092971_2860928604      0          0    1454008249   
1170996282170475597_2603666216      0          0    1453813656   

                                                                    Link  \
UserId                                                                     
1172628647727092971_2860928604  https://www.instagram.com/p/BBGA4rROFTr/   
1170996282170475597_2603666216  https://www.instagram.com/p/BBANuoYEwRN/   

                                  Selfie  BodySnap  Marketing  ProductOnly  \
UserId                                                                       
1172628647727092971_2860928604  0.000499   0.06023   0.006729     0.698443   
1170996282170475597_2603666216  0.008247   0.08745   0.032036     0.759844   

                                NonFashion      Face      Logo  BrandLogo  \
UserId                                                                      
1172628647727092971_2860928604    0.007520  0.012045  0.533595   0.041080   
1170996282170475597_2603666216    0.017295  0.029323  0.936350   0.566902   

                                   Smile   Outdoor  NumberOfPeople  \
UserId                                                               
1172628647727092971_2860928604  0.002225  0.000824        0.214680   
1170996282170475597_2603666216  0.009668  0.020149        0.589774   

                                NumberOfFashionProduct  Anger  Contempt  \
UserId                                                                    
1172628647727092971_2860928604                 2.28591    0.0       0.0   
1170996282170475597_2603666216                 2.14382    0.0       0.0   

                                Disgust  Fear  Happiness  Neutral  Sadness  \
UserId                                                                       
1172628647727092971_2860928604      0.0   0.0        0.0      0.0      0.0   
1170996282170475597_2603666216      0.0   0.0        0.0      0.0      0.0   

                                Surprise  Designer  High street  Mega couture  \
UserId                                                                          
1172628647727092971_2860928604       0.0         0            1             0   
1170996282170475597_2603666216       0.0         1            0             0   

                                Small couture  LikesToF

Drop rows where LikesToFollowers is NAN. Likes and Followers count at both 0.

In [336]:
data_ver2 = data_ver2.dropna()

In [337]:
data_ver2['LikesToFollowers'].isnull().any()

False

In [338]:
X = data_ver2[['Selfie', 'BodySnap', 'Marketing', 'ProductOnly', 'NonFashion',
       'Face', 'Logo', 'BrandLogo', 'Smile', 'Outdoor', 'NumberOfPeople',
       'NumberOfFashionProduct', 'Anger', 'Contempt', 'Disgust', 'Fear',
       'Happiness', 'Neutral', 'Sadness', 'Surprise', 'Designer',
       'High street', 'Mega couture', 'Small couture']]
y = data_ver2['LikesToFollowers']

In [339]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

In [340]:
def standardize(train, test):


    mean = np.mean(train, axis=0)
    std = np.std(train, axis=0)+0.000001

    X_train = (train - mean) / std
    X_test = (test - mean) /std
    return X_train, X_test

In [341]:
X_train, X_test = standardize(X_train, X_test)

In [342]:
X_train

Selfie  BodySnap  Marketing  ProductOnly  \
UserId                                                                       
1171435358474800898_2002493650 -0.257829 -0.825983  -0.527356     1.372176   
1170371593584137509_1495669756 -0.227933  1.804304  -0.477248    -0.933533   
1171553482977030551_2031918563 -0.258684  1.681969  -0.503979    -0.828043   
1171440949699641276_196263166  -0.258893 -0.818154  -0.487886     1.369692   
1171468085370242138_2080653280 -0.254818 -0.749698  -0.520174    -0.840094   
...                                  ...       ...        ...          ...   
1171428953965959033_234068     -0.245324 -0.354788  -0.522946     0.743310   
1168716003269284990_1752602948 -0.228085  1.633713   0.281531    -0.932188   
1171093780916631679_13357       2.325838  0.051541   0.178640    -0.929515   
1170434992837341830_1547609203 -0.253568 -0.803642  -0.485238     1.198756   
1170501340466253691_196666794   5.886708 -0.538507  -0.485020    -0.924765   

                                NonFashion      Face      Logo  BrandLogo  \
UserId                                                                      
1171435358474800898_2002493650   -0.461804 -0.823659  1.134444   2.033331   
1170371593584137509_1495669756   -0.469514  1.625746 -1.086625  -0.791010   
1171553482977030551_2031918563   -0.468880 -0.802391  0.205522  -0.661927   
1171440949699641276_196263166    -0.470000 -0.822627  1.298127   0.349457   
1171468085370242138_2080653280    3.177836 -0.761215  1.321600  -0.689656   
...                                    ...       ...       ...        ...   
1171428953965959033_234068       -0.387432 -0.809811 -0.940217  -0.529932   
1168716003269284990_1752602948   -0.468609  1.623745 -1.207221  -0.734564   
1171093780916631679_13357        -0.356607  1.634504 -0.124527  -0.007779   
1170434992837341830_1547609203   -0.194535 -0.815171  1.257634   1.622415   
1170501340466253691_196666794    -0.410348  1.599063 -1.084209  -0.746408   

                                   Smile   Outdoor  NumberOfPeople  \
UserId                                                               
1171435358474800898_2002493650 -0.599170 -0.495454       -0.919080   
1170371593584137509_1495669756  1.901818  3.064318        0.799990   
1171553482977030551_2031918563 -0.557247 -0.468157       -0.672012   
1171440949699641276_196263166  -0.598006 -0.501364       -1.036417   
1171468085370242138_2080653280 -0.536767 -0.466665       -0.737731   
...                                  ...       ...             ...   
1171428953965959033_234068     -0.563907 -0.457990       -0.715464   
1168716003269284990_1752602948  1.257544  3.072531        1.199481   
1171093780916631679_13357       2.352451 -0.438659        1.766141   
1170434992837341830_1547609203 -0.581371 -0.485659       -0.541101   
1170501340466253691_196666794   1.609982 -0.340738        0.321918   

                                NumberOfFashionProduct     Anger  Contempt  \
UserId                                                                       
1171435358474800898_2002493650               -1.193306 -0.101418 -0.129131   
1170371593584137509_1495669756                1.097663 -0.095371  0.078939   
1171553482977030551_2031918563               -0.517076 -0.101418 -0.129131   
1171440949699641276_196263166                -1.024392 -0.101418 -0.129131   
1171468085370242138_2080653280               -1.352897 -0.101418 -0.129131   
...                                                ...       ...       ...   
1171428953965959033_234068                   -0.902361 -0.101418 -0.129131   
1168716003269284990_1752602948                1.234805  0.079776 -0.099348   
1171093780916631679_13357                     0.153108 -0.088203 -0.116059   
1170434992837341830_1547609203               -0.712909 -0.101418 -0.129131   
1170501340466253691_196666794                -0.606399 -0.101274  3.025473   

                                 Disgust      Fear  Happiness   Neutral  \
UserId          

In [343]:
y_train.head()

UserId
1171435358474800898_2002493650    0.011494
1170371593584137509_1495669756    0.094961
1171553482977030551_2031918563    0.000271
1171440949699641276_196263166     0.000000
1171468085370242138_2080653280    0.000306
Name: LikesToFollowers, dtype: float64

In [350]:
import keras
keras.backend.floatx()
keras.backend.set_floatx('float32')
keras.backend.floatx()

'float32'

In [352]:
from keras.layers.normalization import BatchNormalization
from keras.layers.core import Activation, Dropout
from keras.optimizers import Adadelta
from keras.layers.advanced_activations import LeakyReLU

def our_model_ver2():
    model = Sequential()
    model.add(Dense(units=64, input_dim = 24))
    model.add(LeakyReLU(alpha=0.1))
    model.add(Dropout(0.25))

    model.add(Dense(units=64, input_dim = 24))
    model.add(LeakyReLU(alpha=0.1))
    model.add(Dropout(0.5))
    
    model.add(Dense(units=1))
    model.add(LeakyReLU(alpha=0.1))
    opt = Adadelta(clipnorm=0.3)
    model.compile(loss='mean_squared_error', optimizer=opt,metrics=['accuracy'])
    return model

estimator = KerasRegressor(build_fn=our_model_ver2, epochs=5, batch_size=5, verbose=1)

kfold = KFold(n_splits=10)
results = cross_val_score(estimator, X_train, y_train, cv=kfold)
print("Baseline: %.2f (%.2f) MSE" % (results.mean(), results.std()))

Epoch 1/5
14825/14825 [==============================] - 2s 157us/step - loss: nan - accuracy: 0.0017
Epoch 2/5
14825/14825 [==============================] - 2s 145us/step - loss: nan - accuracy: 0.0000e+00
Epoch 3/5
14825/14825 [==============================] - 2s 145us/step - loss: nan - accuracy: 0.0000e+00
Epoch 4/5
12415/14825 [========================>.....] - ETA: 0s - loss: nan - accuracy: 0.0000e+00- ETA: 0s - loss: nan - accuracy: 0.

KeyboardInterrupt: 